### PR TITLE
docs: add page about cluster-api-provider-vcluster

### LINKED ISSUE #490: Add docs for vcluster cluster api provider
We have a shiny new cluster api provider for vcluster and should mention it within our vcluster docs, so that people know that this possibility exists to run vcluster in a more managed fashion.

### PR DESCRIPTION
**Pull request by matskiv**
**What issue type does this pull request address?** (keep at least one, remove the others) 
/kind documentation

**What does this pull request do? Which issues does it resolve?** (use `resolves #<issue_number>` if possible) 
resolves #490 

**Please provide a short message that should be published in the vcluster release notes**
Added a docs page about [cluster-api-provider-vcluster](https://github.com/loft-sh/cluster-api-provider-vcluster)

### CODE REVIEW

File: docs/pages/operator/cluster-api-provider.mdx [hunk 1]
```diff
@@ -0,0 +1,74 @@
+---
+title: Cluster API Provider vcluster
+sidebar_label: Cluster API Provider vcluster
```
Comment (jschalz) [inline docs/pages/operator/cluster-api-provider.mdx:3]: Please remove "vcluster" from the title and sidebar_title.

File: docs/pages/operator/cluster-api-provider.mdx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+---
+
+[Cluster API (CAPI)](https://cluster-api.sigs.k8s.io/) is a Kubernetes sub-project focused on providing declarative APIs and tooling to simplify provisioning, upgrading, and operating multiple Kubernetes clusters. Kubernetes service providers and distribution maintainers implement CAPI providers which allow users to create Kubernetes clusters of their flavor with the CAPI tooling. For the vcluster project, we have also implemented a CAPI provider - [cluster-api-provider-vcluster](https://github.com/loft-sh/cluster-api-provider-vcluster). Instructions below will describe how to instantiate a vcluster instance using CAPI tooling.
```
Comment (jschalz) [inline docs/pages/operator/cluster-api-provider.mdx:6]: Please change "flavor" to "choice".

File: docs/pages/operator/cluster-api-provider.mdx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+## Prerequisites 
+[clusterctl] CLI tool is used to install the vcluster provider and to generate a custom resource that will serve as a declarative definition of the cluster. It is possible to create the custom resource without clusterctl, but in this guide, we will be using it because it provides access to the templates for the `VCluster` custom resource. Please follow [the official installation instructions for clusterctl](https://cluster-api.sigs.k8s.io/user/quick-start.html#install-clusterctl).
```
Comment (jschalz) [inline docs/pages/operator/cluster-api-provider.mdx:9]: "[clusterctl]" needs a link after it, right?

File: docs/pages/operator/cluster-api-provider.mdx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+You will need cluster-admin permissions in a Kubernetes cluster where you wish to install the CAPI stack with the vcluster provider. This is required because vcluster instances that will be created by the vcluster provider may have arbitrary RBAC permission requirements. The vcluster provider installation with a more limited permission scope will be introduced in a future version.
+
+### Provider configuration - only clusterctl v1.1.4 and earlier
+If you are using clusterctl version 1.1.4 or earlier, you will need to add the vcluster provider to your local clusterctl configuration. To check which version you have installed look for the value of `GitVersion` field in the output of the `clusterctl version` command. You may skip to the next chapter if you are using a newer version.
```
Comment (jschalz) [inline docs/pages/operator/cluster-api-provider.mdx:14]: I'd include an output example of what to look for here, like
```bash
> clusterctl version`
GitVersion=1.1.4      # 1.1.4 or earlier
other output...
```

File: docs/pages/operator/cluster-api-provider.mdx [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+Open your local clusterctl configuration file `~/.cluster-api/clusterctl.yaml` and add vcluster to the list of provider as follows:
+
+```yaml
+providers:
+  - name: vcluster
+    url: https://github.com/loft-sh/cluster-api-provider-vcluster/releases/latest/infrastructure-components.yaml
+    type: InfrastructureProvider                                                           
+```
+
+Verify that the vcluster provider was added successfully using the `clusterctl config repositories` command, e.g.:
```
Comment (jschalz) [inline docs/pages/operator/cluster-api-provider.mdx:25]: I like this verification command!

File: docs/pages/operator/cluster-api-provider.mdx [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+```shell
+clusterctl init --infrastructure vcluster
+```
+
+The `"Your management cluster has been initialized successfully!"` message should eventually appear in the output.
+
+## Creating a vcluster
+We will use the clusterctl CLI tool to generate the Kubernetes resource definitions needed for vcluster creation by the CAPI provider. The output is in YAML format, and it can be saved into a file to be used later, or it can be forwarded in the command line into the kubectl command to immediately apply the resources in the cluster.  
+
+A vcluster instance created via vcluster provider is configured using clusterctl parameters and environment variables - CHART_NAME, CHART_REPO, CHART_VERSION, VCLUSTER_HOST, VCLUSTER_PORT, and HELM_VALUES.
```
Comment (jschalz) [inline docs/pages/operator/cluster-api-provider.mdx:42]: I'd change the format of the variables here to either a different format to differentiate them. If the variables are described somewhere else, I'd use option 1. If the variables don't have descriptions somewhere else, I'd use Option 2 or 3.

Option 1: modify text formatting
"A vcluster instance created via vcluster provider is configured using clusterctl parameters and environment variables: `CHART_NAME`, `CHART_REPO`, `CHART_VERSION`, `VCLUSTER_HOST`, `VCLUSTER_PORT`, and `HELM_VALUES`."

Option 2: table (also provides info about the variables)
| Variable | Description | Type | Default Value | Required/Optional |
| :---: | :--- | :---: | :---: | :---: |
| CHART_NAME | description of CHART_NAME | string | default value here | required |
| etc | etc | etc | etc | etc |

Option 3: plain list (also provides more info about the variables)
"A vcluster instance created via vcluster provider is configured using clusterctl parameters and environment variables: 
- `CHART_NAME` (required/optional) (type): description
- `CHART_REPO` (required/optional) (type): description
- `CHART_VERSION` (required/optional) (type): description
- `VCLUSTER_HOST` (required/optional) (type): description
- `VCLUSTER_PORT` (required/optional) (type): description
- `HELM_VALUES` (required/optional) (type): description"

File: docs/pages/operator/cluster-api-provider.mdx [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+
+We will define some configuration elements(CLUSTER_NAME and CLUSTER_NAMESPACE) as environment variables for easier reuse across the commands. These will be used together with some of the environment variables mentioned above to produce a manifest defining your vcluster instance and apply it in the host cluster:
```
Comment (jschalz) [inline docs/pages/operator/cluster-api-provider.mdx:44]: Please add a space between "elements" and "(CLUSTER_NAME"

File: docs/pages/operator/cluster-api-provider.mdx [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+:::tip
+You can populate the HELM_VALUES variable from a `values.yaml` file using this command:
+```export HELM_VALUES=$(cat values.yaml | sed -z 's/\n/\\n/g')```
+:::
+
+Next, we need to wait until vcluster custom resource reports ready status:
+```shell
+kubectl wait --for=condition=ready vcluster -n $CLUSTER_NAMESPACE $CLUSTER_NAME 
+```
+The cluster is ready to be used once the command above exists.
```
Comment (jschalz) [inline docs/pages/operator/cluster-api-provider.mdx:65]: Please change "exists" to "exits"!

File: docs/pages/operator/cluster-api-provider.mdx [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+
+### Connecting to your vcluster
+There are multiple methods for connecting to your vcluster instance, and they are described on the [Accessing vcluster page](./external-access). If the documentation instructs you to update your values.yaml file, you will need to update the HELM_VALUES environment variable, execute the clusterctl generate cluster again, and apply the updated output to you cluster. Alternatively, you can edit the `.spec.helmRelease.values` field of the vcluster custom resource using `kubectl edit vcluster -n $CLUSTER_NAMESPACE $CLUSTER_NAME ` command, but be mindful of the formatting of the value, the string should be a valid YAML file, with the expected indentation for the various helm values. The vcluster provider is watching for the changes made to vcluster custom resource, and it will automatically update the vcluster instance when changes are detected.
```
Comment (jschalz) [inline docs/pages/operator/cluster-api-provider.mdx:68]: Should the "accessing vcluster" link be (./external-access.mdx) or (./accessing-vcluster.mdx)?

Comment (jschalz) [inline docs/pages/operator/cluster-api-provider.mdx:68]: I'd add a line break between "If the documentation instructs you to update your values.yaml file, you will need to update the HELM_VALUES environment variable, execute the clusterctl generate cluster again, and apply the updated output to you cluster." and "Alternatively, you can edit the `.spec.helmRelease.values` field of the vcluster custom resource using `kubectl edit vcluster -n $CLUSTER_NAMESPACE $CLUSTER_NAME ` command, but be mindful of the formatting of the value, the string should be a valid YAML file, with the expected indentation for the various helm values." Together it gets a bit wordy. I'd also split the "Alternatively" sentence.

"There are multiple methods for connecting to your vcluster instance, and they are described on the [Accessing vcluster page](./external-access). If the documentation instructs you to update your values.yaml file, you will need to update the HELM_VALUES environment variable, execute the clusterctl generate cluster again, and apply the updated output to you cluster. 

Alternatively, you can edit the `.spec.helmRelease.values` field of the vcluster custom resource using `kubectl edit vcluster -n $CLUSTER_NAMESPACE $CLUSTER_NAME ` command. Be mindful of the formatting of the value: the string should be a valid YAML file, with the expected indentation for the various helm values. The vcluster provider is watching for the changes made to vcluster custom resources, and it will automatically update the vcluster instance when changes are detected."

**Review by jschalz — DISMISSED** [review]
Overall great tutorial! Some sentences got a bit long, but otherwise the grammar was pretty good!

I added a couple data formatting and spelling changes!

**Review by FabianKramm — APPROVED** [review]
